modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/validator/AMImageValidatorImpl.java

```diff
@@ -75,19 +81,112 @@ public boolean isValid(FileVersion fileVersion) {
 			return false;
 		}
 
+		if (!_isFileVersionStoredMetadataSupported(fileVersion)) {
+			return false;
+		}
+
+		return true;
+	}
+
+	private boolean _checkDDMFormFieldsMapValue(
+		List<DDMFormFieldValue> ddmFormFieldValues,
+		long imageToolImageMaxValue) {
+
+		DDMFormFieldValue ddmFormFieldValue = ddmFormFieldValues.get(0);
+
+		Value value = ddmFormFieldValue.getValue();
+
+		Long length = Long.valueOf(value.getString(value.getDefaultLocale()));
+
+		if (length <= 0) {
+			return true;
+		}
+		else if (length >= imageToolImageMaxValue) {
+			return false;
+		}
+
 		return true;
 	}
 
-	@Activate
-	@Modified
-	protected void activate(Map<String, Object> properties) {
-		_amImageConfiguration = ConfigurableUtil.createConfigurable(
-			AMImageConfiguration.class, properties);
+	private boolean _isFileVersionStoredMetadataSupported(
+		FileVersion fileVersion) {
+
+		List<DDMStructure> ddmStructures =
+			_ddmStructureManager.getClassStructures(
+				fileVersion.getCompanyId(),
+				_portal.getClassNameId(RawMetadataProcessor.class),
+				DDMStructureManager.STRUCTURE_COMPARATOR_STRUCTURE_KEY);
+
+		for (DDMStructure ddmStructure : ddmStructures) {
+			DLFileEntryMetadata fileEntryMetadata =
+				_dlFileEntryMetadataLocalService.fetchFileEntryMetadata(
+					ddmStructure.getStructureId(),
+					fileVersion.getFileVersionId());
+
+			if (fileEntryMetadata != null) {
+				try {
+					DDMFormValues ddmFormValues =
+						_storageEngine.getDDMFormValues(
+							fileEntryMetadata.getDDMStorageId());
+
+					Map<String, List<DDMFormFieldValue>> ddmFormFieldValuesMap =
+						ddmFormValues.getDDMFormFieldValuesMap(true);
+
+					for (Map.Entry<String, List<DDMFormFieldValue>> entry :
+							ddmFormFieldValuesMap.entrySet()) {
+
+						if (Objects.equals(
```

Comment:
Why not use or (||) instead of two if clauses?

Reply:
Props values passed to the _checkDDMFormFieldsMapValue method are different if both clauses.